File: cpp/soma_matrizes.cpp
#include <iostream>

using namespace std;

int main() {

    int M, N;

    cout << "Quantas linhas vai ter cada matriz? ";
    cin >> M;
    cout << "Quantas colunas vai ter cada matriz? ";
    cin >> N;

    int a[M][N], b[M][N], c[M][N];

    cout << "Digite os valores da matriz A:" << endl;

    for (int i = 0; i < M; i++) {
        for (int j = 0; j < N; j++) {
            cout << "Elemento [" << i << "," << j << "]: ";
            cin >> a[i][j];
        }
    }

    cout << "Digite os valores da matriz B:" << endl;

    for (int i = 0; i < M; i++) {
        for (int j = 0; j < N; j++) {
            cout << "Elemento [" << i << "," << j << "]: ";
            cin >> b[i][j];
        }
    }

    for (int i = 0; i < M; i++) {
        for (int j = 0; j < N; j++) {
            c[i][j] = a[i][j] + b[i][j];
        }
    }

    cout << "MATRIZ SOMA:" << endl;

    for (int i = 0; i < M; i++) {
        for (int j = 0; j < N; j++) {
            cout << c[i][j] << " ";
        }
        cout << endl;
    }

    return 0;
}

File: python/soma_matrizes.py
M: int
N: int

M = int(input("Quantas linhas vai ter cada matriz? "))
N = int(input("Quantas colunas vai ter cada matriz? "))

a: [[int]] = [[0 for x in range(N)] for x in range(M)]
b: [[int]] = [[0 for x in range(N)] for x in range(M)]
c: [[int]] = [[0 for x in range(N)] for x in range(M)]

print("Digite os valores da matriz A:")

for i in range(M):
    for j in range(N):
        a[i][j] = int(input(f"Elemento [{i},{j}]: "))

print("Digite os valores da matriz B:")

for i in range(M):
    for j in range(N):
        b[i][j] = int(input(f"Elemento [{i},{j}]: "))

for i in range(M):
    for j in range(N):
        c[i][j] = a[i][j] + b[i][j]

print("MATRIZ SOMA:")

for i in range(M):
    for j in range(N):
        print(f"{c[i][j]} ", end="")
    print()
